utils/hare-fetch-fids

```diff
@@ -80,54 +84,131 @@ def node_names(data: List) -> List[str]:
 
-        for epdata in data:
-            result = re.match(regex_ep, epdata['key'])
-            if not result:
-                continue
+    for key in data:
+        match_result = re.match(regex, key['key'])
+        if not match_result:
+            continue
+        return key['value']
 
-            keys.append(KeyData(match_result.group(0), epdata['value']))
-            break
-    if not keys:
-        raise RuntimeError(
-            'No ioservice, confd, or any motr client found in the cluster')
-    return keys
+    return None
+
+
+def generate_node_uuid():
+    p = Program(['uuidgen', '--time'])
+    return Executor().run(p)
+
+
+def get_confd_svc(node: Node, data: List) -> Dict[str, Any]:
+    return {'ha_ep': node.ha_ep,
+            'conf_xc': '/etc/motr/confd.xc',
+            'uuid': node.uuid}
+
+
+def get_io_svc(fid: str, node: Node,
+               data: List) -> Dict[str, Any]:
+    svc: Dict[str, Any] = {}
+    svc = {'ha_ep': node.ha_ep,
+           'uuid': node.uuid}
+
+    metadata = get_ios_metadata(fid, node.name, data)
+    if metadata:
+        svc.update({'be_seg_path': metadata})
+
+    return svc
+
+
+def get_ios_metadata(fid: str, node_name: str, data: List):
+    # Key e.g. '"key": "m0conf/nodes/localhost/processes/12/meta_data",
+    #           "value": "/dev/sda"'
+    regex = re.compile(f'^m0conf\\/nodes\\/{node_name}\\/processes\\/{fid}\\'
+                       '/meta_data')
+
+    for key in data:
+        match_result = re.match(regex, key['key'])
+        if not match_result:
+            continue
+        return key['value']
+    return None
+
+
+def get_profile_fid(data: List):
+    # Key e.g. "m0conf/profiles/0x7000000000000001:0x59"
+    regex = re.compile('^m0conf\\/profiles\\/.*$')
+
+    for key in data:
+        match_result = re.match(regex, key['key'])
+        if not match_result:
+            continue
+        parts = key['key'].split('/')
+        return parts[2]
+    return None
+
+
+def get_motr_client_svc(node: Node, data: List):
+    return {'ha_ep': node.ha_ep,
+            'profile_fid': node.profile_fid}
 
 
 def add_service_to_node(nodes: List[Node], node_name: str, svc_name: str,
-                        fid: str, ep: str) -> None:
+                        data: List, process_fidk: str, ep: str) -> None:
+    fid = to_fid(ObjT.PROCESS.value, int(process_fidk))
     for node in nodes:
         if node.name == node_name:
-            node.add_service(svc_name, fid, ep)
-
-
-def get_service_for_node(nodes: List[Node], node_name: str,
-                         service: str, path: str) -> Optional[List[Process]]:
+            svc: Dict[str, Any] = {}
+            svc.update({'name': svc_name, 'fid': fid, 'ep': ep})
+            if svc_name == 'confd':
+                svc.update(get_confd_svc(node, data))
+            elif svc_name == 'ioservice':
+                svc.update(get_io_svc(process_fidk, node, data))
+            elif svc_name != 'hax':
```

Comment:
Not sure I understand, so if service is not hax and ioservices and confd then it is expected to be a motr_client service, we are then trying to fetch motr_client service information but `get_motr_client_svc()` returns `ha endpoint` and `profile_fid`, so as a reader I don't really understand exactly what information we are populating.